### PR TITLE
coll: Convert buffer swap macro to function

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

Other support for collective GPU buffer replacement is implemented
using functions. Convert this macro to a function for consistency.

## Expected Impact

just code consistency. no functional change.

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [x] Passes warning tests
* [x] Passes all tests
* [x] Add comments such that someone without knowledge of the code could understand
* [x] You or your company has a signed contributor's agreement on file with Argonne
* [x] For non-Argonne authors, request an explicit comment from your companies PR approval manager

### CODE REVIEW

**Comment by raffenet** [comment]
test:mpich/ch4/most

File: src/mpi/coll/src/coll_impl.c [hunk 1]
```diff
@@ -256,3 +256,27 @@ void MPIR_Coll_host_buffer_free(void *host_sendbuf, void *host_recvbuf)
     MPL_free(host_sendbuf);
     MPL_free(host_recvbuf);
 }
+
+void MPII_Coll_host_buffer_swap_back(void *host_sendbuf, void *host_recvbuf, void *in_recvbuf,
+                                     MPI_Aint count, MPI_Datatype datatype, MPIR_Request ** request)
```
Comment (hzhou) [inline src/mpi/coll/src/coll_impl.c:261]: Is it necessary to make `request` INOUT?

Comment (raffenet) [reply src/mpi/coll/src/coll_impl.c:261]: It's not. I kept the changes to a minimum for copy/paste reasons, but I can update this part. It's a worthwhile improvement.

File: src/mpi/coll/src/coll_impl.c [hunk 2]
```diff
@@ -256,3 +256,27 @@ void MPIR_Coll_host_buffer_free(void *host_sendbuf, void *host_recvbuf)
     MPL_free(host_sendbuf);
```
Comment (hzhou) [inline src/mpi/coll/src/coll_impl.c:256]: The other two functions are using `MPIR_` prefix. I think we should make it consistent, all use `MPIR_` or all use `MPII_`

**Comment by raffenet** [comment]
test:mpich/ch4/most

File: src/mpi/coll/src/coll_impl.c (hunk 2 above)
Comment (raffenet) [reply src/mpi/coll/src/coll_impl.c:256]: Done.

File: src/mpi/coll/src/coll_impl.c (hunk 1 above)
Comment (raffenet) [reply src/mpi/coll/src/coll_impl.c]: Done.

**Review by hzhou — APPROVED** [review]
Looks good as long as tests are clear

**Comment by raffenet** [comment]
Getting out of memory errors in this round of Jenkins results. Seems unrelated to the changes in this PR.

**Comment by raffenet** [comment]
test:mpich/ch4/most

**Comment by raffenet** [comment]
test:mpich/ch4/most

**Comment by raffenet** [comment]
test:mpich/ch4/ofi

**Comment by raffenet** [comment]
test:mpich/ch4/ucx

**Pull request merged**